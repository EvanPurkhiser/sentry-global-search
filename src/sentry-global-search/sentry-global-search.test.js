import SentryGlobalSearch from './sentry-global-search';

const config = ['docs', 'develop', 'blog', 'help-center'];

describe('Search', () => {
  test('requires configuration', () => {
    expect(() => {
      new SentryGlobalSearch();
    }).toThrowErrorMatchingSnapshot();
  });

  test('requires configuration as array', () => {
    expect(() => {
      const search = new SentryGlobalSearch(
        'docs',
        'develop',
        'blog',
        'help-center'
      );
    }).toThrowErrorMatchingSnapshot();
  });

  test('rejects unknown sites', () => {
    expect(() => {
      const search = new SentryGlobalSearch(['foobar']);
    }).toThrowErrorMatchingSnapshot();
  });

  test('returns nothing for an empty query', async () => {
    const search = new SentryGlobalSearch(config);
    const results = await search.query();
    expect(results).toEqual([]);
  });

  test('queries', async () => {
    const search = new SentryGlobalSearch(config);
    const results = await search.query('react');
    expect(results).toMatchSnapshot();
  });
});
