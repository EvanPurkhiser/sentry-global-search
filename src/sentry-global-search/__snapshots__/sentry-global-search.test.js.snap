// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search queries 1`] = `
Array [
  Object {
    "hits": Array [
      Object {
        "context": Object {
          "context1": "Platforms > JavaScript > React",
        },
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "docs",
        "text": "MockedContent",
        "title": "<mark>React</mark>",
        "url": "https://docs.sentry.io/platforms/javascript/react/",
      },
      Object {
        "context": Object {
          "context1": "Platforms > JavaScript > React",
        },
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "docs",
        "text": "MockedContent",
        "title": "<mark>React</mark>",
        "url": "https://docs.sentry.io/platforms/javascript/react/",
      },
    ],
    "name": "Documentation",
    "site": "docs",
  },
  Object {
    "hits": Array [
      Object {
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "develop",
        "text": "Mocked excerpt",
        "title": "<mark>React</mark>",
        "url": "https://develop.sentry.dev/platforms/javascript/react/",
      },
      Object {
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "develop",
        "text": "Mocked excerpt",
        "title": "<mark>React</mark>",
        "url": "https://develop.sentry.dev/platforms/javascript/react/",
      },
    ],
    "name": "Developer Documentation",
    "site": "develop",
  },
  Object {
    "hits": Array [
      Object {
        "context": Object {
          "context1": "Mocked hit",
        },
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "blog",
        "text": "Mocked text",
        "title": "Mocked section",
        "url": "https://blog.sentry.io/platforms/javascript/react/",
      },
      Object {
        "context": Object {
          "context1": "Mocked hit",
        },
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "blog",
        "text": "Mocked text",
        "title": "Mocked section",
        "url": "https://blog.sentry.io/platforms/javascript/react/#mocked-anchor",
      },
    ],
    "name": "Blog Posts",
    "site": "blog",
  },
  Object {
    "hits": Array [
      Object {
        "context": Object {
          "context1": "Mocked full_path",
        },
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "help-center",
        "text": "Mocked body_safe",
        "title": "<mark>React</mark>",
        "url": "https://help.sentry.io/hc/en-us/articles/undefined",
      },
      Object {
        "context": Object {
          "context1": "Mocked full_path",
        },
        "id": "cd5528ad-935b-5a33-b25e-4f945638551b",
        "site": "help-center",
        "text": "Mocked body_safe",
        "title": "<mark>React</mark>",
        "url": "https://help.sentry.io/hc/en-us/articles/undefined",
      },
    ],
    "name": "Help Center",
    "site": "help-center",
  },
]
`;

exports[`Search rejects unknown sites 1`] = `"SentryGlobalSearchError: unknown site \\"foobar\\" in config.include"`;

exports[`Search requires configuration 1`] = `"SentryGlobalSearchError: SentryGlobalSearch must be initialized with an array that includes one or more of: docs, develop, help-center, blog."`;

exports[`Search requires configuration as array 1`] = `"SentryGlobalSearchError: SentryGlobalSearch must be initialized with an array that includes one or more of: docs, develop, help-center, blog."`;
